### DISCUSSION TITLE
Button pressed and released

### DISCUSSION THREAD

**Discussion by cbradburne** [MyreMylar/pygame_gui#198] [Ideas]
It would be great if you could make the button pressed event actually "go" on the press and not the release. Also, great to have a "button released" event too.

I had to use PyGame button (not PyGame_GUI) to "fix" a couple of buttons... Not ideal.

**Comment by MyreMylar** [comment]
There is an event for when a button is first pushed down, though it's not documented yet:

UI_BUTTON_START_PRESS

should do what you want.

**Reply by cbradburne** [reply]
Amazing, thank you! 

I guess I can use the "UI_BUTTON_PRESSED" for the button release / has finished being pressed.
